src/posthog-surveys.ts

```diff
@@ -131,13 +135,39 @@ export function getNextSurveyStep(
     return nextQuestionIndex
 }
 
+function defaultMatchType(matchType?: SurveyMatchType): SurveyMatchType {
+    return matchType ?? 'icontains'
+}
+
 // use urlMatchType to validate url condition, fallback to contains for backwards compatibility
 export function doesSurveyUrlMatch(survey: Survey): boolean {
     if (!survey.conditions?.url) {
         return true
     }
+    // if we dont know the url, assume it is not a match
+    const href = assignableWindow?.location?.href
```

Comment:
is it ok to use `assignableWindow` here?

Reply:
I think if you're getting the `location.href`, which is a default field in the `window` object, you should probably just use the `window.location.href`. but @pauldambra you're my local posthog-js expert, what do you think?

Reply:
`window` does not exist in that context (or at least there's no code using it) and the code uses `assignableWindow` instead so I guess this is the right thing to do? but yeah @pauldambra \o/

Reply:
i think it doesn't matter here too much... `assignableWindow` just does some helpful type fangling and in this instance you don't (i don't think) need it

there is a `window` that should be being exported alongside `assignableWindow`

i guess it's more expressive to use that since you're not attempting to assign

Reply:
https://github.com/PostHog/posthog-js/blob/2a06f91fe83d905f6728d2d44cc65c83a5cb5259/src/utils/globals.ts#L113